### PR TITLE
Move to "secrets" list instead of field property.

### PR DESCRIPTION
**Pull request by underdarknl**
### Changes
Fix secrets like we also handle required fields.

### CODE REVIEW

File: rocky/katalogus/views/mixins.py [hunk 1]
```diff
@@ -71,11 +71,12 @@ def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
     def is_required_field(self, field: str) -> bool:
+        """Check whether this field should be required, defaults to False."""        
         return self.plugin_schema and field in self.plugin_schema.get("required", [])
 
     def is_secret_field(self, fields: dict) -> bool:
```
Comment (zcrt) [inline rocky/katalogus/views/mixins.py]:
```suggestion
    def is_secret_field(self, field: str) -> bool:
```

File: rocky/katalogus/views/mixins.py [hunk 2]
```diff
@@ -71,11 +71,12 @@ def dispatch(self, request, *args, **kwargs):
...
         """Check whether this field should be secret, defaults to False."""
-        return fields.get("hide_value_in_interface", False)
+        return self.plugin_schema and field in self.plugin_schema.get("secret", [])
```
Comment (zcrt) [inline rocky/katalogus/views/mixins.py:79]: This is now almost a copy of `is_required_field`, maybe a separate function to call would be cleaner

**Review by zcrt — CHANGES_REQUESTED** [review]

File: rocky/katalogus/views/mixins.py (hunk 2 above)
Comment (underdarknl) [reply rocky/katalogus/views/mixins.py:79]: It is, and yes, maybe we can refactor this, but lets keep this simple for now.

**Comment by zcrt** [comment]
Related: https://github.com/json-schema-org/json-schema-vocabularies/issues/6

**Pull request merged**